aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/RetryHandler.kt

```diff
@@ -0,0 +1,52 @@
+package com.amplifyframework.datastore.syncengine
```

Comment:
Currently it looks like this is a pretty specific retry implementation - it only handles retrying an AppSync sync request, rather than just any general request or operation.   Similar to the `OnSuccessConsumer`, since it is so closely related to the `SyncProcessor`, I think it should probably be the same language as the `SyncProcessor` (Java).

That said, it would be awesome to see this be a little more generic. Is there a way we could adapt it to take any generic RxJava `Single`?  Then, we could potentially use it in other places in the app.  In that case, it would be a standalone utility that I think we could justify using Kotlin for.